install on Fedora with QEMU mode

**Describe the bug**
I downloaded the source with git clone and ran the make distrib command, however, QEMU mode didn't compile. So I went to manually run the build script, but it also errors out.

**To Reproduce**
Steps to reproduce the behavior:
1. `git clone https://github.com/AFLplusplus/AFLplusplus`
2. `sudo make distrib`
3. `sudo make install`
4. `sudo ./build_qemu_support.sh`

**Screen output/Screenshots**
```
FAILED: [code=1] libqemu-x86_64-linux-user.fa.p/accel_tcg_cpu-exec.c.o 
cc -Ilibqemu-x86_64-linux-user.fa.p -I. -I.. -Itarget/i386 -I../target/i386 -I../linux-user/host/x86_64 -Ilinux-user -I../linux-user -Ilinux-user/x86_64 -I../linux-user/x86_64 -I../capstone/include/capstone -Itrace -Iqapi -Iui/shader -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-6 -fdiagnostics-color=auto -Wall -Winvalid-pch -std=gnu99 -O2 -isystem /home/andromeda/AFLplusplus/qemu_mode/qemuafl/linux-headers -isystem linux-headers -iquote . -iquote /home/andromeda/AFLplusplus/qemu_mode/qemuafl -iquote /home/andromeda/AFLplusplus/qemu_mode/qemuafl/include -iquote /home/andromeda/AFLplusplus/qemu_mode/qemuafl/disas/libvixl -iquote /home/andromeda/AFLplusplus/qemu_mode/qemuafl/tcg/i386 -iquote /home/andromeda/AFLplusplus/qemu_mode/qemuafl/accel/tcg -pthread -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -m64 -mcx16 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -Wstrict-prototypes -Wredundant-decls -Wundef -Wwrite-strings -Wmissing-prototypes -fno-strict-aliasing -fno-common -fwrapv -Wno-int-conversion -Wold-style-declaration -Wold-style-definition -Wtype-limits -Wformat-security -Wformat-y2k -Winit-self -Wignored-qualifiers -Wempty-body -Wnested-externs -Wendif-labels -Wexpansion-to-defined -Wimplicit-fallthrough=2 -Wno-missing-include-dirs -Wno-shift-negative-value -Wno-psabi -Wno-unused-function -fPIE -isystem../linux-headers -isystemlinux-headers -DNEED_CPU_H '-DCONFIG_TARGET="x86_64-linux-user-config-target.h"' '-DCONFIG_DEVICES="x86_64-linux-user-config-devices.h"' -MD -MQ libqemu-x86_64-linux-user.fa.p/accel_tcg_cpu-exec.c.o -MF libqemu-x86_64-linux-user.fa.p/accel_tcg_cpu-exec.c.o.d -o libqemu-x86_64-linux-user.fa.p/accel_tcg_cpu-exec.c.o -c ../accel/tcg/cpu-exec.c
../accel/tcg/cpu-exec.c:288:33: error: initialization of ‘uint64_t *’ {aka ‘long unsigned int *’} from incompatible pointer type ‘unsigned char *’ [-Wincompatible-pointer-types]
  288 | static uint64_t *ijon_max_ptr = dummy;
      |                                 ^~~~~
../accel/tcg/cpu-exec.c:586:10: warning: no previous prototype for ‘ijon_simple_hash’ [-Wmissing-prototypes]
  586 | uint64_t ijon_simple_hash(uint64_t x) {
      |          ^~~~~~~~~~~~~~~~
../accel/tcg/cpu-exec.c:593:10: warning: no previous prototype for ‘ijon_hashint’ [-Wmissing-prototypes]
  593 | uint32_t ijon_hashint(uint32_t old, uint32_t val) {
      |          ^~~~~~~~~~~~
../accel/tcg/cpu-exec.c:621:10: warning: no previous prototype for ‘ijon_hashstr’ [-Wmissing-prototypes]
  621 | uint32_t ijon_hashstr(uint32_t old, char *val) {
      |          ^~~~~~~~~~~~
../accel/tcg/cpu-exec.c:683:6: warning: no previous prototype for ‘ijon_max_variadic’ [-Wmissing-prototypes]
  683 | void ijon_max_variadic(uint32_t addr, ...) {
      |      ^~~~~~~~~~~~~~~~~
../accel/tcg/cpu-exec.c:711:6: warning: no previous prototype for ‘ijon_min_variadic’ [-Wmissing-prototypes]
  711 | void ijon_min_variadic(uint32_t addr, ...) {
      |      ^~~~~~~~~~~~~~~~~
../accel/tcg/cpu-exec.c:756:10: warning: no previous prototype for ‘ijon_strdist’ [-Wmissing-prototypes]
  756 | uint32_t ijon_strdist(char *a, char *b) {
      |          ^~~~~~~~~~~~
../accel/tcg/cpu-exec.c: In function ‘afl_forkserver’:
../accel/tcg/cpu-exec.c:1307:30: warning: pointer targets in passing argument 2 of ‘waitpid’ differ in signedness [-Wpointer-sign]
 1307 |       if (waitpid(child_pid, &status, 0) < 0) exit(8);
      |                              ^~~~~~~
      |                              |
      |                              u32 * {aka unsigned int *}
In file included from /home/andromeda/AFLplusplus/qemu_mode/qemuafl/include/qemu/osdep.h:108,
                 from ../accel/tcg/cpu-exec.c:20:
/usr/include/sys/wait.h:106:45: note: expected ‘int *’ but argument is of type ‘u32 *’ {aka ‘unsigned int *’}
  106 | extern __pid_t waitpid (__pid_t __pid, int *__stat_loc, int __options);
      |                                        ~~~~~^~~~~~~~~~
../accel/tcg/cpu-exec.c:1358:28: warning: pointer targets in passing argument 2 of ‘waitpid’ differ in signedness [-Wpointer-sign]
 1358 |     if (waitpid(child_pid, &status, is_persistent ? WUNTRACED : 0) < 0) exit(6);
      |                            ^~~~~~~
      |                            |
      |                            u32 * {aka unsigned int *}
/usr/include/sys/wait.h:106:45: note: expected ‘int *’ but argument is of type ‘u32 *’ {aka ‘unsigned int *’}
  106 | extern __pid_t waitpid (__pid_t __pid, int *__stat_loc, int __options);
      |                                        ~~~~~^~~~~~~~~~
```

**Additional context**
Using Fedora 43, kde

--- a/accel/tcg/cpu-exec.c
+++ b/accel/tcg/cpu-exec.c
@@ -583,14 +583,16 @@ void ijon_dispatch(IJON v, uint32_t addr, u64 val) {
 #include <time.h>
 
 /* Supporting hash functions */
-uint64_t ijon_simple_hash(uint64_t x) {
+static uint64_t ijon_simple_hash(uint64_t x)
+{
 
   const uint64_t golden_ratio = 0x9E3779B97F4A7C15ULL;
   return x * golden_ratio;
 
 }
 
-uint32_t ijon_hashint(uint32_t old, uint32_t val) {
+static uint32_t ijon_hashint(uint32_t old, uint32_t val)
+{
 
   // PERFECT HASH: Bit-interleaving approach for coordinate pairs
   // Guarantees no hash collisions for coordinates < 65536
@@ -618,7 +620,8 @@ uint32_t ijon_hashint(uint32_t old, uint32_t val) {
 
 }
 
-uint32_t ijon_hashstr(uint32_t old, char *val) {
+static uint32_t ijon_hashstr(uint32_t old, char *val)
+{
 
   return ijon_hashmem(old, val, strlen(val));
 
@@ -680,7 +683,8 @@ void ijon_inc(uint32_t loc_addr, uint32_t val) {
 }
 
 /* Variadic runtime functions */
-void ijon_max_variadic(uint32_t addr, ...) {
+static void ijon_max_variadic(uint32_t addr, ...)
+{
 
   va_list args;
   va_start(args, addr);
@@ -708,7 +712,8 @@ void ijon_max_variadic(uint32_t addr, ...) {
 
 }
 
-void ijon_min_variadic(uint32_t addr, ...) {
+static void ijon_min_variadic(uint32_t addr, ...)
+{
 
   va_list args;
   va_start(args, addr);
@@ -753,7 +758,8 @@ void ijon_reset_state(uint32_t addr, u64 val) {
 
 /* String and memory distance functions */
 
-uint32_t ijon_strdist(char *a, char *b) {
+static uint32_t ijon_strdist(char *a, char *b)
+{
 
   if (!a && !b) return 0;
   if (!a) return strlen(b);

--- a/accel/tcg/cpu-exec.c
+++ b/accel/tcg/cpu-exec.c
@@ -1310,7 +1310,7 @@ void afl_forkserver(CPUState *cpu) {
     if (child_stopped && was_killed) {
 
       child_stopped = 0;
-      if (waitpid(child_pid, &status, 0) < 0) exit(8);
+      if (waitpid(child_pid, (int *)&status, 0) < 0) exit(8);
 
     }
 
@@ -1361,7 +1361,7 @@ void afl_forkserver(CPUState *cpu) {
 
     /* Get and relay exit status to parent. */
 
-    if (waitpid(child_pid, &status, is_persistent ? WUNTRACED : 0) < 0) exit(6);
+    if (waitpid(child_pid, (int *)&status, is_persistent ? WUNTRACED : 0) < 0) exit(6);
 
     /* In persistent mode, the child stops itself with SIGSTOP to indicate
        a successful run. In this case, we want to wake it up without forking

--- a/gdbstub.c
+++ b/gdbstub.c
@@ -65,12 +65,12 @@ static int phy_memory_mode;
 
 
 void (*signal_handler)(int);
-void set_signal_callback(void (*sg)(int)){
+static void set_signal_callback(void (*sg)(int)){
     signal_handler = sg;
 }
 
-int target_memory_rw_debug(CPUState *cpu, target_ulong addr,
-                                         uint8_t *buf, int len, bool is_write)
+static int target_memory_rw_debug(CPUState *cpu, target_ulong addr,
+                                  uint8_t *buf, int len, bool is_write)
 {
     CPUClass *cc;
 
@@ -383,7 +383,7 @@ typedef struct GDBState {
 static int sstep_flags = SSTEP_ENABLE|SSTEP_NOIRQ|SSTEP_NOTIMER;
 
 /* Retrieves flags for single step mode. */
-int get_sstep_flags(void)
+static int get_sstep_flags(void)
 {
     /*
      * In replay mode all events written into the log should be replayed.
@@ -473,7 +473,7 @@ int use_gdb_syscalls(void)
 }
 
 /* Resume execution.  */
-void gdb_continue(void)
+static void gdb_continue(void)
 {
 
 #ifdef CONFIG_USER_ONLY
@@ -926,7 +926,7 @@ static const char *get_feature_xml(const char *p, const char **newp,
     return name ? xml_builtin[i][1] : NULL;
 }
 
-int gdb_read_register(CPUState *cpu, GByteArray *buf, int reg)
+static int gdb_read_register(CPUState *cpu, GByteArray *buf, int reg)
 {
     CPUClass *cc = CPU_GET_CLASS(cpu);
     CPUArchState *env = cpu->env_ptr;
@@ -944,7 +944,7 @@ int gdb_read_register(CPUState *cpu, GByteArray *buf, int reg)
     return 0;
 }
 
-int gdb_write_register(CPUState *cpu, uint8_t *mem_buf, int reg)
+static int gdb_write_register(CPUState *cpu, uint8_t *mem_buf, int reg)
 {
     CPUClass *cc = CPU_GET_CLASS(cpu);
     CPUArchState *env = cpu->env_ptr;
@@ -1023,7 +1023,7 @@ static inline int xlat_gdb_type(CPUState *cpu, int gdbtype)
 }
 #endif
 
-int gdb_breakpoint_insert(int type, target_ulong addr, target_ulong len)
+static int gdb_breakpoint_insert(int type, target_ulong addr, target_ulong len)
 {
     CPUState *cpu;
     int err = 0;
@@ -1060,7 +1060,7 @@ int gdb_breakpoint_insert(int type, target_ulong addr, target_ulong len)
     }
 }
 
-int gdb_breakpoint_remove(int type, target_ulong addr, target_ulong len)
+static int gdb_breakpoint_remove(int type, target_ulong addr, target_ulong len)
 {
     CPUState *cpu;
     int err = 0;
@@ -1128,7 +1128,7 @@ static void gdb_breakpoint_remove_all(void)
     }
 }
 
-void gdb_set_cpu_pc(target_ulong pc)
+static void gdb_set_cpu_pc(target_ulong pc)
 {
     CPUState *cpu = gdbserver_state.c_cpu;
 
@@ -3215,7 +3215,7 @@ void gdb_signalled(CPUArchState *env, int sig)
     put_packet(buf);
 }
 
-void gdb_accept_init(int fd)
+static void gdb_accept_init(int fd)
 {
     init_gdbserver_state();
     create_default_process(&gdbserver_state);

